File: Algorithms/src/1404. Number of Steps to Reduce a Number in Binary Representation to One/Code/solution.go
func numSteps(s string) int {
    steps := 0
    carry := 0
    
    // Traverse from right to left (ignore first bit)
    for i := len(s) - 1; i > 0; i-- {
        bit := int(s[i]-'0') + carry
        
        if bit == 1 {
            // Odd case
            steps += 2
            carry = 1
        } else {
            // Even case
            steps += 1
        }
    }
    
    return steps + carry
}

File: Algorithms/src/1404. Number of Steps to Reduce a Number in Binary Representation to One/Code/solution.java
class Solution {
    public int numSteps(String s) {
        int steps = 0;
        int carry = 0;

        // Traverse from right to left (ignore first bit)
        for (int i = s.length() - 1; i > 0; i--) {
            int bit = (s.charAt(i) - '0') + carry;

            if (bit == 1) {
                // Odd case
                steps += 2;
                carry = 1;
            } else {
                // Even case
                steps += 1;
            }
        }

        return steps + carry;
    }
}